File: class-diagram-generator/sample-projects/maven-multi-jdk17/support/src/main/java/com/example/catalog/support/CatalogSupport.java
package com.example.catalog.support;

/**
 * 서비스 계층이 재사용하는 보조 기능 계약이다.
 */
public interface CatalogSupport {
    /**
     * 상품 식별자를 보기 좋은 형식으로 정리한다.
     *
     * @param sku 원본 상품 식별자
     * @return 정리된 상품 식별자
     */
    String formatSku(String sku);
}

File: class-diagram-generator/sample-projects/gradle-multi-kotlin-jdk21/support/src/main/kotlin/com/example/catalog/support/CatalogSupport.kt
package com.example.catalog.support

/** 카탈로그 등급 */
enum class CatalogTier {
    BASIC,
    PREMIUM
}

/** 모듈 간 공유되는 스냅샷 모델 */
data class CatalogSnapshot(
    val key: String,
    val value: String,
    val tier: CatalogTier
)

/** 지원 모듈 계약 */
interface CatalogSupport {
    fun current(): List<CatalogSnapshot>
}

/** 키 생성 유틸리티 */
object CatalogKeyFactory {
    fun keyOf(id: Long): String = "catalog-$id"
}
